### ISSUE TITLE
clang: Assembler args are undocumented (-Wa, -Xassembler)

### ISSUE THREAD

**Issue by ekliptik** [llvm/llvm-project#84633]
it seems that some clang options are also valid `-Wa` options. Is there anything beyond the source code of `CollectArgsForIntegratedAssembler` that lists which ones are available? For example, [GNU as has a man page](https://linux.die.net/man/1/as). Meanwhile, `clang -Wa,--help` errors out with "error: no input files" just like `clang -Wl,--help -fuse-ld=lld` but unlike `clang -Wl,--help`.

**Comment by brad0** [comment]
cc @MaskRay

**Comment by MaskRay** [comment]
Indeed, there is no documentation except the code (`CollectArgsForIntegratedAssembler` as you mentioned):)

---

Clang uses the integrated assembler for most targets. Unlike `-fno-integrated-as`, which passes `-Wa,xxx` to the assembler (usually gas), the integrated assembler handles a few most used `-Wa,` options. In the wild, `-Wa,` options are used scarcely, so our limited handling turns out to be sufficient. We also set a pretty high bar to add new options. (I recently came up https://github.com/llvm/llvm-project/pull/82188 (`-Wa,--fdpic`), but now I realize that it is not a good idea and GCC should be fixed instead)
